chain/syncer.go

```diff
@@ -5,18 +5,23 @@ package chain
 
 import (
 	"context"
+	"errors"
+
+	"github.com/ava-labs/hypersdk/internal/validity_window"
 )
 
+var ErrInvalidExecutionBlockType = errors.New("invalid execution block type")
+
 // Syncer marks sequential blocks as accepted until it has observed a full validity window
 // and signals to the caller that it can begin processing blocks from that block forward.
 type Syncer struct {
-	chainIndex         ChainIndex
-	timeValidityWindow *TimeValidityWindow
+	chainIndex         validity_window.ChainIndex[*Transaction]
+	timeValidityWindow validity_window.TimeValidityWindow[*Transaction]
```

Comment:
Can we move the syncer into the validity window package as well since it is required to find the point where we can safely start to apply de-duplication?

Reply:
done